### DISCUSSION TITLE
Smart Motion Detect

### DISCUSSION THREAD

**Discussion by DuelCoreF18** [briis/unifiprotect#216] [Ideas]
Hi Briis,
First off, I really appreciate your work. This component is integral to my home setup.

The one enhancement I would really like would be to have a binary_sensor.smart_detect_DEVICE utilizing the smart detection of people and cars in the newer g4 devices in addition to binary_sensor.motion_DEVICE. 

I have some G4 bullets and a G4 doorbell. My use case is that when my doorbell detect motion it turns on the front light. I don't have a large front yard and my motion zone is only from the house to the sidewalk. Unfortunately when I car passes my house at night the light and shadows moving across my yard trip the motion detection. I'm still playing with my motion zone and sensitivity, but being able to use smart detection would yield much better results. Would also help keeping the lights from turning on thanks to motion detection in the backyard from small animals, but that's not nearly as bad as the cars tripping the porch light.

**Comment by DuelCoreF18** [comment]
This might be a nevermind. I think I found the relevant part of the wiki regarding this.

**Reply by briis** [reply]
See my reply in #171
